test/libyul/yulOptimizerTests/redundantAssignEliminator/for_continue.yul

```diff
@@ -0,0 +1,33 @@
+{
+	// Cannot be removed, because we might run the loop only once
+	let x := 1
+	for { } calldataload(0) { }
+	{
+		x := 2 // Will not be removed as if-condition can be false.
+		if callvalue() {
+			x := 3
```

Comment:
This could actually be removed.

Reply:
Ah, I see why it is still there: We only copy out the state at the continue statement, we do not move it. So the control flow that "skipes" the continue statement is also taken into account.

Reply:
The if statement already branches into two states, so I might want to try starting with a "fresh" state just after the continue.

Reply:
Yep that did it. It caused no other changes in the test expectations.

Reply:
thanks!